src/components/application_manager/test/commands/hmi/get_urls_test.cc

```diff
@@ -234,7 +235,137 @@ TEST_F(GetUrlsTest, ProcessPolicyServiceURLs_SUCCESS) {
+  (*command_msg)[am::strings::msg_params][am::strings::number] = "123";
+  (*command_msg)[am::strings::params][am::strings::connection_key] =
+      kConnectionKey;
+  (*command_msg)[am::strings::msg_params][am::hmi_request::service] =
+      kPolicyService;
+
+  ON_CALL(mock_app_manager_, event_dispatcher())
+      .WillByDefault(ReturnRef(mock_event_dispatcher_));
+
+  RequestFromHMIPtr command(CreateCommand<GetUrls>(command_msg));
+
+  EXPECT_CALL(mock_app_manager_, GetPolicyHandler()).Times(3);
+  EXPECT_CALL(policy_handler_, PolicyEnabled()).WillOnce(Return(true));
+
+  EndpointUrls endpoints_;
+  EndpointData data(kDefaultUrl);
+  endpoints_.push_back(data);
+
+  ON_CALL(policy_handler_, GetServiceUrls(kPolicyService, _))
+      .WillByDefault(SetArgReferee<1>(endpoints_));
+  EXPECT_CALL(policy_handler_, GetServiceUrls(kPolicyService, _));
+
+  EXPECT_CALL(policy_handler_, GetAppIdForSending())
+      .WillOnce(Return(kInvalidAppId_));
+
+  EXPECT_CALL(mock_app_manager_, ManageHMICommand(command_msg))
+      .WillRepeatedly(Return(true));
+
+  EXPECT_CALL(mock_app_manager_, application(kInvalidAppId_)).Times(0);
+
+  command->Run();
+}
+TEST_F(GetUrlsTest, ProcessPolicyServiceURLs_ApplicationIsNotValid_UNSUCCESS) {
+  MessageSharedPtr command_msg(
+      CreateMessage(NsSmartDeviceLink::NsSmartObjects::SmartType_Map));
+  (*command_msg)[am::strings::msg_params][am::strings::number] = "123";
+  (*command_msg)[am::strings::params][am::strings::connection_key] =
+      kConnectionKey;
+  (*command_msg)[am::strings::msg_params][am::hmi_request::service] =
+      kPolicyService;
+
+  ON_CALL(mock_app_manager_, event_dispatcher())
+      .WillByDefault(ReturnRef(mock_event_dispatcher_));
+
+  RequestFromHMIPtr command(CreateCommand<GetUrls>(command_msg));
+
+  EXPECT_CALL(mock_app_manager_, GetPolicyHandler()).Times(3);
+  EXPECT_CALL(policy_handler_, PolicyEnabled()).WillOnce(Return(true));
+
+  EndpointUrls endpoints_;
+  EndpointData data(kDefaultUrl);
+  endpoints_.push_back(data);
+
+  ON_CALL(policy_handler_, GetServiceUrls(kPolicyService, _))
+      .WillByDefault(SetArgReferee<1>(endpoints_));
+  EXPECT_CALL(policy_handler_, GetServiceUrls(kPolicyService, _));
+
+  MockAppPtr mock_app;
+
+  EXPECT_CALL(policy_handler_, GetAppIdForSending())
+      .WillOnce(Return(kAppIdForSending));
+
+  EXPECT_CALL(mock_app_manager_, application(kAppIdForSending))
+      .WillOnce(Return(mock_app));
+
+  EXPECT_CALL(mock_app_manager_, ManageHMICommand(command_msg))
+      .WillRepeatedly(Return(true));
+
+  command->Run();
+
+  EXPECT_EQ((*command_msg)[strings::params][strings::message_type].asInt(),
+            am::MessageType::kResponse);
+  EXPECT_EQ((*command_msg)[strings::params][am::hmi_response::code].asInt(),
+            Common_Result::DATA_NOT_AVAILABLE);
+}
+TEST_F(GetUrlsTest, ProcessPolicyServiceURLs_FoundURLForApplication_SUCCESS) {
+  MessageSharedPtr command_msg(
+      CreateMessage(NsSmartDeviceLink::NsSmartObjects::SmartType_Map));
```

Comment:
@dcherniev Maybe you need move this logic to separete method, in order to avoid dublicate code?